src/Functions/fromModifiedJulianDay.cpp

```diff
@@ -232,7 +232,7 @@ namespace DB
         static constexpr auto name = "fromModifiedJulianDayOrNull";
     };
 
-    void registerFunctionFromModifiedJulianDay(FunctionFactory & factory)
+    __attribute__((used)) __attribute__((used)) void registerFunctionFromModifiedJulianDay(FunctionFactory & factory)
```

Comment:
Duplicate.

`__attribute__((used)) ` is not obvious for common developers. One may ask why we need such a GNU attribute extension when developing a function. I'd suggest we implement a macro `REGISTER_FUNCTION(name)` which does the same thing but comes with descriptive comments. It also helps avoid this "duplication" issue.

Reply:
How about add a macro `#define ATTR_IS_USED __attribute__((used))`? My impression was that the `REGISTER_FUNCTION()` macro makes the code looks like C according to @alexey-milovidov .

Reply:
A bit confused, `SymbolIndex` also plays with ELF directly?

Reply:
Would agree with @amosbird here ... a REGISTER_FUNCTION() macro would be nice (and afair, an earlier state already had a macro).

Reply:
@amosbird  If macro is not the concern but the platform dependent part is, I propose to use the platform dependent implementation if it's supported, and fall back to iterator over symbols to load all functions if not. How do you think?

Reply:
@amosbird updated to use the FUNCTION_REGISTER() macro, and fallback to iterator over symbols if __start and __stop is not supported. Please help to review again, thanks.

Reply:
Why do we need `HAS_ELF_START_STOP_SYMBOL` branch? The original suggestion is to avoid manipulating ELF sectors by only iterating symbols directly.

Reply:
Both methods are supported with the change, and fallback to iterating over symbols if platform doesn't have the `__start` / `__stop` support. I can change to only iterating over symbols if that's what you want.

Reply:
removed